File: exercise/ts/day10/tests/delivery.spec.ts
import * as fs from "node:fs";
import {Building} from "../src/delivery/building";
import * as path from "node:path";

describe('Building', () => {
    test.each([
        ["1", 0],
        ["2", 3],
        ["3", -1],
        ["4", 53],
        ["5", -3],
        ["6", 2920]
    ])('returns the correct floor number based on instructions in file %s', (fileName, expectedFloor) => {
        const instructions = fs.readFileSync(path.join(__dirname,`deliveries/${fileName}.txt`), 'utf-8');
        const result = Building.whichFloor(instructions);
        expect(result).toBe(expectedFloor);
    });
});

File: exercise/scala/day10/src/test/scala/DeliverySpec.scala
import delivery.Building
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers
import org.scalatest.prop.TableDrivenPropertyChecks.*
import org.scalatest.prop.TableFor2

import scala.io.Source

class DeliverySpec extends AnyFunSuite with Matchers {

  import DeliverySpec.*

  val testData: TableFor2[String, Int] = Table(
    ("fileName", "expectedFloor"),
    ("1.txt", 0),
    ("2.txt", 3),
    ("3.txt", -1),
    ("4.txt", 53),
    ("5.txt", -3),
    ("6.txt", 2920)
  )

  test("return floor number based on instructions") {
    forAll(testData) { (fileName: String, expectedFloor: Int) =>
      val instructions = loadInstructionsFromFile(fileName)
      Building.whichFloor(instructions) should be(expectedFloor)
    }
  }
}

object DeliverySpec {
  def loadInstructionsFromFile(input: String): String = {
    val source = Source.fromResource(input)
    try source.mkString finally source.close()
  }
}
